scripts/ci/operator.sh

```diff
@@ -0,0 +1,188 @@
+  kubectl -n "$ns" edit "deploy/istio-operator"
+}
+
+logs_operator() {
+  echo "Display the Operator logs"
+  echo
+
+  local ns=${ns:-openshift-operators}
+  kubectl -n "$ns" logs deploy/istio-operator "${args[*]}"
+}
+
+restart_operator() {
+  echo "Restart the operator."
+  echo
+
+  local ns=${ns:-openshift-operators}
+  echo "Restart operator by deleting the Pod"
+  kubectl -n "$ns" delete pod -l name=istio-operator --force --grace-period 0
+}
+
+# we can remove this function in testing 2.4
+wait_operator() {
+  echo "Temporary workaround for running operator 2.3.x"
+  echo
+
+  echo -n "Waiting for Jaeger operator deployment to be created..."
+  while ! kubectl get deployment -n openshift-operators -o name | grep jaeger >& /dev/null ; do echo -n '.'; sleep 1; done
+  echo "done."
+  jaeger_deployment="$(kubectl get deployment -n openshift-operators -o name | grep jaeger)"
+
+  echo -n "Waiting for Kiali operator deployment to be created..."
+  while ! kubectl get deployment -n openshift-operators -o name | grep kiali >& /dev/null ; do echo -n '.'; sleep 1; done
+  echo "done."
+  kiali_deployment="$(kubectl get deployment -n openshift-operators -o name | grep kiali)"
+
+  echo -n "Waiting for Service Mesh operator deployment to be created..."
+  while ! kubectl get deployment -n openshift-operators -o name | grep istio >& /dev/null ; do echo -n '.'; sleep 1; done
+  echo "done."
+  servicemesh_deployment="$(kubectl get deployment -n openshift-operators -o name | grep istio)"
+
+  echo "Waiting for CRDs to be established."
+  for crd in servicemeshcontrolplanes.maistra.io servicemeshmemberrolls.maistra.io jaegers.jaegertracing.io
+  do
+    echo -n "Waiting for CRD [${crd}]..."
+    while ! kubectl get crd "${crd}" >& /dev/null ; do echo -n '.'; sleep 1; done
+    kubectl wait --for condition=established crd/"${crd}"
+  done
+
+  echo "Waiting for operator deployments to start..."
+  for op in "${servicemesh_deployment}" "${jaeger_deployment}"
+  do
+    echo -n "Waiting for ${op} to be ready..."
+    readyReplicas="0"
+    while [ "$?" != "0" -o "$readyReplicas" == "0" ]
+    do
+      sleep 1
+      echo -n '.'
+      readyReplicas="$(kubectl get ${op} -n openshift-operators -o jsonpath='{.status.readyReplicas}' 2> /dev/null)"
+    done
+    echo "done."
+  done
+
+  echo "Wait for the servicemesh validating webhook to be created."
+  while [ "$(kubectl get validatingwebhookconfigurations -o name | grep servicemesh)" == "" ]; do echo -n '.'; sleep 5; done
+  echo "done."
+
+  echo "Wait for the servicemesh mutating webhook to be created."
+  while [ "$(kubectl get mutatingwebhookconfigurations -o name | grep servicemesh)" == "" ]; do echo -n '.'; sleep 5; done
+  echo "done."
+
+  echo "Wait for the smcp.mutation.maistra.io webhook to be created."
+  while [ "$(kubectl get mutatingwebhookconfigurations -o name | grep smcp.mutation.maistra.io)" == "" ]; do echo -n '.'; sleep 5; done
+  echo "done."
```

Comment:
I don't think this does anything. The webhook entry in the mutatingwebhookconfiguration object will always be present. Instead, you should actually call the webhook here, not check the mutatingwebhookconfiguration resource.

Reply:
thanks, I updated that check. It waits for the maistra-admission-controller curl call now.